module/VuFind/src/VuFind/Search/History.php

```diff
@@ -136,7 +138,12 @@ public function getScheduleOptions()
         if (!($this->config->Account->schedule_searches ?? false)) {
             return [];
         }
-        return $this->config->Account->scheduled_search_frequencies
-            ?? [0 => 'schedule_none', 1 => 'schedule_daily', 7 => 'schedule_weekly'];
+        if ($this->config->Account->scheduled_search_frequencies instanceof Config) {
```

Comment:
Is it ok to check instanceof with a potentially missing element? I suppose this works most of the time as the Account section is usually defined, but if it wasn't, wouldn't this balk? So it would be safer to move this check inside the isset block below?

Reply:
If the account section isn't set, then the check for scheduled_searches will evaluate to false, and the code will never get this far. I think that makes this safe, though perhaps in need of a clarifying comment. However, your observation also reminds me that our test coverage for this method is not complete, since otherwise this bug would never have gotten through in the first place... thus, I'll expand the tests today to make sure all the edge cases are checked, and make adjustments as needed.

Reply:
You're right. There's still a chance that a later change could introduce a bug since it's sort of a side-effect, but I suppose it's fine.